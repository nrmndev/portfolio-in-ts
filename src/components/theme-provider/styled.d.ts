// styled.d.ts
import "styled-components";
interface IPalette {
  main: string;
  contrastText: string;
}
declare module "styled-components" {
  export interface DefaultTheme {
    theme: string;
    backgroundColor: string;
    backgroundColorFlat: string;
    backgroundColorFlatAlt: string;
    backgroundColorGradient: string;
    backgroundColorGradientAlt: string;
    borderColor: string;
    boxShadow: string;
    boxShadowInset: string;
    color: string;
    colorAsGradientClip: string;
    colorPrimary: string;
    colorWithContrast: string;
    colorContrast: string;
    colorPallete1: string;
  }
}
// borderRadius: string;
// palette: {
//   common: {
//     black: string;
//     white: string;
//   };
//   primary: IPalette;
//   secondary: IPalette;
// };
