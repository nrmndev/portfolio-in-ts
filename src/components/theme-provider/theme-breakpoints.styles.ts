export const sizes = {
  mobileS: "320px",
  mobileM: "375px",
  mobileL: "425px",
  tablet: "768px",
  laptop: "1024px",
  laptopL: "1440px",
  desktop: "2560px",
};

export const fontSize = {
  H1: "4.25rem;",
  H2: "2.5rem",
  H3: "2rem",
  H4: "1.5rem",
  H5: "1.35rem",
  H6: "1rem",
  P: "1rem",
};
