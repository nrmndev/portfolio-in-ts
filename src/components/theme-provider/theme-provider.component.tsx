import { ThemeProvider } from "styled-components";
import { useAppSelector } from "../../store/hooks/typedhooks";
import { uiTheme } from "../../store/ui/ui-selector";

type ReduxThemeProviderProps = {
  children: React.ReactNode;
};
const ReduxThemeProvider = ({ children }: ReduxThemeProviderProps) => {
  const theme = useAppSelector(uiTheme);

  return <ThemeProvider theme={{ theme: theme }}>{children}</ThemeProvider>;
};

export default ReduxThemeProvider;
