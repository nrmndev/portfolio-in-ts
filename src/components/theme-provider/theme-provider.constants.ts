const gutter = {
  NONE: "0",
  XS: "1rem",
  SM: "1.5rem",
  MD: "2rem",
  LG: "3rem",
  XL: "4rem",
  XXL: "6rem",
};

export const themeUtilities = {
  gutterX: {
    NONE: `padding-top: ${gutter.NONE}; padding-bottom: ${gutter.NONE}`,
    XS: `padding-top: ${gutter.XS}; padding-bottom: ${gutter.XS}`,
    SM: `padding-top: ${gutter.SM}; padding-bottom: ${gutter.SM}`,
    MD: `padding-top: ${gutter.MD}; padding-bottom: ${gutter.MD}`,
    LG: `padding-top: ${gutter.LG}; padding-bottom: ${gutter.LG}`,
    XL: `padding-top: ${gutter.XL}; padding-bottom: ${gutter.XL}`,
    XXL: `padding-top: ${gutter.XXL}; padding-bottom: ${gutter.XXL}`,
  },
};
