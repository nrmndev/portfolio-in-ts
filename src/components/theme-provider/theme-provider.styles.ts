import styled from "styled-components";
import theme from "styled-theming";

export const themedMainNavTextColor = theme("theme", {
  light: "#4a4752",
  dark: "#43cea2",
});
export const themedButtonTextColor = theme("theme", {
  light: "#000",
  dark: "#43cea2",
});

export const themedParagraphColor = theme("theme", {
  light: "#777",
  dark: "#c4cfde",
});

export const themedHeadingColor = theme("theme", {
  light: "#777",
  dark: "#fff",
});

export const themedGradientBackgroundColor = theme("theme", {
  light: "linear-gradient(165deg,#fff 50%,#edebeb 80%)",
  dark: "linear-gradient(145deg,#1e2024,#23272b)",
});

export const themedBackgroundColor = theme("theme", {
  light: "#f8f8f8",
  dark: "#26282c",
});

export const themedBoxShadow = theme("theme", {
  light: "2px 2px 15px #d8d5d5",
  dark: "0 20px 20px #1c1d21",
});
