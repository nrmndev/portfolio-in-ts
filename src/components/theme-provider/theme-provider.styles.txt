// import styled from "styled-components";
// import theme from "styled-theming";

// export const themedBackgroundClip =
//   "linear-gradient(30deg, rgb(67, 206, 162) 0%, rgb(24, 90, 157))";

// export const themedStaticPrimaryColor = theme("theme", {
//   light: "#43cea2",
//   dark: "#43cea2",
// });

// export const themedMainNavTextColor = theme("theme", {
//   light: "#4a4752",
//   dark: "#43cea2",
// });

// // Button color: theme
// export const themedButtonTextColor = theme("theme", {
//   light: "#43cea2",
//   dark: "#43cea2",
// });

// // Button color: theme
// export const themedButtonTextColor2 = theme("theme", {
//   light: "#fff",
//   dark: "#43cea2",
// });

// // Body/P/Li color: [theme]
// export const themedParagraphColor = theme("theme", {
//   light: "#3c3e41",
//   dark: "#c4cfde",
// });

// // H1-H6 color: [theme]
// export const themedHeadingColor = theme("theme", {
//   light: "#777",
//   dark: "#c4cfde",
// });

// export const themedFormBorder = theme("theme", {
//   light: "#dce1e4",
//   dark: "#191b1e",
// });
// //#dce1e4
// export const themedFormBackground = theme("theme", {
//   light: "#FFFFFF",
//   dark: "#191b1e",
// });
// export const themedFormBoxShadow = theme("theme", {
//   light: "none",
//   dark: "1px 4px 2px -3px rgba(0, 0, 0, 0.7) inset,-1px -3px 3px -2px rgba(255, 255, 255, 0.2) inset",
// });

// // export const themedGradientBackgroundColor = theme("theme", {
// //   light: "linear-gradient(165deg,#fff 50%,#edebeb 80%)",
// //   dark: "linear-gradient(145deg,#1e2024,#23272b)",
// // });

// export const themedBackgroundColor = theme("theme", {
//   light: "#ECF0F3",
//   dark: "#212428",
// });

// //Timeline border-color: [theme]
// export const themedTimelineColor = theme("theme", {
//   light: "#d9dee2",
//   dark: "#17191c",
// });

// //Gradient background: [theme]
// export const themedGradientBackgroundColor = theme("theme", {
//   light: "linear-gradient(145deg, #e2e8ec, #ffffff)",
//   dark: "linear-gradient(145deg,#1e2024,#23272b)",
// });

// //Hover Gradient background: [theme]
// //light: "linear-gradient(145deg, #6a67ce, #fc636b)",
// export const themedGradientBackgroundColor2 = theme("theme", {
//   light: "linear-gradient(221deg, #43cea2 0%, #185a9d)",
//   dark: "linear-gradient(to right bottom, #212428, #16181c)",
// });

// //Box Shadow background: [theme]
// export const themedBoxShadow = theme("theme", {
//   light: "5px 5px 15px #D1D9E6, -5px -5px 15px #ffffff",
//   dark: "10px 10px 19px #1c1e22, -10px -10px 19px #262a2e",
// });

// // export const themedBoxShadowV2 = theme("theme", {
// //   dark: "linear-gradient(to right bottom, #212428, #16181c)"
// // })
// //dark: "0 20px 20px #1c1d21",

// //IMG backface bg: [theme]
// export const themedImageBackfaceBG = theme("theme", {
//   light: "none",
//   dark: "#FFF",
// });

// //Border top-color: [theme]
// export const themedBorderColor = theme("theme", {
//   light: "#dce1e4",
//   dark: "#121415",
// });
