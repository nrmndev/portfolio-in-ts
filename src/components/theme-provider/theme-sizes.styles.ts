const gutter = {
  NONE: "0",
  XS: "1rem",
  SM: "1.5rem",
  MD: "2rem",
  LG: "3rem",
  XL: "4rem",
  XXL: "6rem",
};

export const themedGutterX = {
  NONE: `padding-top: ${gutter.NONE}; padding-bottom: ${gutter.NONE}`,
  XS: `padding-top: ${gutter.XS}; padding-bottom: ${gutter.XS}`,
  SM: `padding-top: ${gutter.SM}; padding-bottom: ${gutter.SM}`,
  MD: `padding-top: ${gutter.MD}; padding-bottom: ${gutter.MD}`,
  LG: `padding-top: ${gutter.LG}; padding-bottom: ${gutter.LG}`,
  XL: `padding-top: ${gutter.XL}; padding-bottom: ${gutter.XL}`,
  XXL: `padding-top: ${gutter.XXL}; padding-bottom: ${gutter.XXL}`,
};

export const themedGutterY = {
  NONE: `padding-left: ${gutter.NONE}; padding-right: ${gutter.NONE}`,
  XS: `padding-left: ${gutter.XS}; padding-right: ${gutter.XS}`,
  SM: `padding-left: ${gutter.SM}; padding-right: ${gutter.SM}`,
  MD: `padding-left: ${gutter.MD}; padding-right: ${gutter.MD}`,
  LG: `padding-left: ${gutter.LG}; padding-right: ${gutter.LG}`,
  XL: `padding-left: ${gutter.XL}; padding-right: ${gutter.XL}`,
  XXL: `padding-left: ${gutter.XXL}; padding-right: ${gutter.XXL}`,
};
