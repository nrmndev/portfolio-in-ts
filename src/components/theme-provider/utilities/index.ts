export {
  UTIL_PADDING as PADDING,
  UTIL_H_PADDING as H_PADDING,
  UTIL_V_PADDING as V_PADDING,
  handlePadding,
  type PaddingTypes,
} from "./padding";

export {
  UTIL_TEXT_AS as TEXT_AS,
  UTIL_FONT_SIZE_AS as FONT_SIZE_AS,
  UTIL_TEXT_TRANSFORM as TEXT_TRANSFORM,
  UTIL_TEXT_ALIGN as TEXT_ALIGN,
  handleTextAlign,
  handleFontSize,
  handleTextTransform,
} from "./typography";

export { UTIL_TEXT_COLOR as TEXT_COLOR, handleTextColor } from "./color";

export {
  UTIL_BUTTON_VARIANT as BUTTON_VARIANT,
  UTIL_BUTTON_SIZE as BUTTON_SIZE,
  handleButtonSize,
} from "./button";
export {
  UTIL_CARD_VARIANT as CARD_VARIANT,
  UTIL_CARD_PADDING as CARD_PADDING,
} from "./card";
export {
  UTIL_BADGE_VARIANT as BADGE_VARIANT,
  handleBadgeVariant,
} from "./badge";
export { UTIL_LOADING_SKELETON_TYPE as LOADING_SKELETON_TYPE } from "./loading-skeleton";
export { UTIL_IMAGE_VARIANT as IMAGE_VARIANT } from "./image";
export { UTIL_BORDER_WIDTH as BORDER_WIDTH } from "./border";
export { UTIL_GAP_SEPARATOR_VARIANT as GAP_SEPARATOR_VARIANT } from "./gap-separator";
export { UTIL_INPUT_TYPE as INPUT_TYPE } from "./input";
export { UTIL_BOX_SHADOW as BOX_SHADOW, handleBoxShadow } from "./box-shadow";
export {
  UTIL_BACKGROUND_COLOR as BACKGROUND_COLOR,
  handleBackgroundColor,
} from "./background-color";
