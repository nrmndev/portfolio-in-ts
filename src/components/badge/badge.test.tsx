import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Badge from "./badge.component";
import "jest-styled-components";
import { ThemeProviderTestWrapper } from "../theme-provider/theme-provider.component";
import { BADGE_VARIANT } from "../theme-provider/utilities";
import { renderWithProviders } from "../../utils/test/test-utils";
import { useStore } from "react-redux";

describe("Badge Component", () => {
  jest.mock("axios");
  test("renders Badge with children props", () => {
    const store = useStore();
    render(
      <ThemeProviderTestWrapper>
        <Badge data-testid="badge" variant={BADGE_VARIANT.PRIMARY}>
          Hello
        </Badge>
      </ThemeProviderTestWrapper>
    );

    renderWithProviders(
      <Badge data-testid="badge" variant={BADGE_VARIANT.PRIMARY}>
        Hello
      </Badge>,
      { store }
    );
    const hello = screen.getByText("Hello");

    expect(hello).toBeInTheDocument();
  });

  test("renders Badge with variant = PRIMARY", () => {
    const tree = renderer
      .create(
        <ThemeProviderTestWrapper>
          <Badge data-testid="badge" variant={BADGE_VARIANT.PRIMARY}>
            Hello
          </Badge>
        </ThemeProviderTestWrapper>
      )
      .toJSON();
    expect(tree).toHaveStyleRule(`background: #43cea2`);

    expect(tree).toMatchSnapshot();
  });

  test("renders Badge with variant = SECONDARY", () => {
    const tree = renderer
      .create(
        <ThemeProviderTestWrapper>
          <Badge data-testid="badge" variant={BADGE_VARIANT.SECONDARY}>
            Hello
          </Badge>
        </ThemeProviderTestWrapper>
      )
      .toJSON();
    expect(tree).toHaveStyleRule(`background: #185a9d`);

    expect(tree).toMatchSnapshot();
  });

  test("renders Badge with variant = SUCCESS", () => {
    const tree = renderer
      .create(
        <ThemeProviderTestWrapper>
          <Badge data-testid="badge" variant={BADGE_VARIANT.SUCCESS}>
            Hello
          </Badge>
        </ThemeProviderTestWrapper>
      )
      .toJSON();
    expect(tree).toHaveStyleRule(`background: green`);

    expect(tree).toMatchSnapshot();
  });

  test("renders Badge with variant = WARNING", () => {
    const tree = renderer
      .create(
        <ThemeProviderTestWrapper>
          <Badge data-testid="badge" variant={BADGE_VARIANT.WARNING}>
            Hello
          </Badge>
        </ThemeProviderTestWrapper>
      )
      .toJSON();
    expect(tree).toHaveStyleRule(`background: yellow`);

    expect(tree).toMatchSnapshot();
  });
});
