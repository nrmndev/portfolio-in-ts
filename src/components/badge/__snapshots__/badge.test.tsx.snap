// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component renders Badge with variant = PRIMARY 1`] = `
.c0 {
  font-size: 0.7rem;
  padding: 0.3rem 0.6rem;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.375rem;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  background: #43cea2;
  color: #c4cfde;
}

<span
  className="c0"
>
  Hello
</span>
`;

exports[`Badge Component renders Badge with variant = SECONDARY 1`] = `
.c0 {
  font-size: 0.7rem;
  padding: 0.3rem 0.6rem;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.375rem;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  background: #43cea2;
  color: #c4cfde;
}

<span
  className="c0"
>
  Hello
</span>
`;

exports[`Badge Component renders Badge with variant = SUCCESS 1`] = `
.c0 {
  font-size: 0.7rem;
  padding: 0.3rem 0.6rem;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.375rem;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  background: green;
  color: #c4cfde;
}

<span
  className="c0"
>
  Hello
</span>
`;

exports[`Badge Component renders Badge with variant = WARNING 1`] = `
.c0 {
  font-size: 0.7rem;
  padding: 0.3rem 0.6rem;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.375rem;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  background: yellow;
  color: #c4cfde;
}

<span
  className="c0"
>
  Hello
</span>
`;
